src/main/java/lrbg/codriver/command/DeadlineCommand.java

```diff
@@ -0,0 +1,55 @@
+package lrbg.codriver.command;
+
+import lrbg.codriver.data.Deadline;
+import lrbg.codriver.data.exception.CoDriverException;
+import lrbg.codriver.data.Task;
+import lrbg.codriver.data.TaskList;
+import lrbg.codriver.storage.Storage;
+import lrbg.codriver.ui.Ui;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a command to add a deadline task to the task list.
+ */
+public class DeadlineCommand extends Command {
+    /** The description of the deadline task. */
+    private final String description;
+    /** The date of the deadline task. */
+    private final LocalDate date;
+
+    /**
+     * Creates a new DeadlineCommand with the given description and date.
+     * @param description The description of the deadline task.
+     * @param date The date of the deadline task.
+     */
+    public DeadlineCommand(String description, LocalDate date) {
+        this.description = description;
+        this.date = date;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws CoDriverException {
```

Comment:
I wonder if theres a better name for CoDriverException? CoDriverException doesnt tell me much about what kind of error is happening